### ISSUE TITLE
eliminate ATVocabularyManager

### ISSUE THREAD

**Issue by paregorios** [isawnyu/pleiades-gazetteer#371]
Finish removing dependency on ATVocabularyManager and migrate any remaining vocabs to PleiadesSettings (to include surfacing the new vocab in HTML at /vocabularies/ and to include exporting same to JSON).

Remove ATVocabularyManager from the Pleiades buildout. Hunt down and exterminate related cruft.

**Comment by paregorios** [comment]
Blocker for #374 (upgrade site to Plone 5)

**Comment by paregorios** [comment]
I believe that these are the vocabularies that are still using ATVM:

 - "association certainty" (association-certainty)
 - "attestation confidence" (attestation-confidence)
 - "language and script" (ancient-name-languages)
 - "name type" (name-types)
 - "name accuracy" (name-accuracy)
 - "name completeness" (name-completeness)

**Comment by alecpm** [comment]
I've created PRs in [`pleiades.vocabularies`](https://github.com/isawnyu/pleiades-vocabularies/pull/7) and [`Products.PleiadesEntities`](https://github.com/isawnyu/PleiadesEntity/pull/39), merged those branches to staging, deployed and run the migrations. All the vocabs on staging should now be registry based vocabularies, but otherwise be identical to the ATVM vocabularies. This will require relatively thorough testing because these vocabularies are used extensively, including in the exports.

Additionally, once this is approved I plan on making minor changes to a number of other repositories to remove their dependencies on ATVM. Those are all trivial test or package install time dependencies, so the changes should not have any impact on functionality. I have already tested those changes locally, but didn't want to make a dozen more PRs.

**Comment by paregorios** [comment]
Below is the testing regime I propose to follow:

## Verify I can see ~~and edit~~ all expected vocabularies via the control panel
 - [x] Time Periods
 - [x] Feature (or Place) Categories
 - [x] Archaeological Remains
 - [x] Location Categories
 - [x] Connection Types (aka Relationship Types)
 - [x] Association Certainty
 - [x] Attestation Confidence
 - [x] Language and Script
 - [x] Name Types
 - [x] Name Accuracy
 - [x] Name Completeness

## Verify I can __edit__ all expected vocabularies via the control panel
 - [x] Time Periods
 - [x] Feature (or Place) Categories
 - [x] Archaeological Remains
 - [x] Location Categories
 - [x] Connection Types (aka Relationship Types)
 - [x] Association Certainty
 - [x] Attestation Confidence
 - [x] Language and Script
 - [x] Name Types
 - [x] Name Accuracy
 - [x] Name Completeness

## Verify I can see HTML serializations of all the vocabularies at ```/vocabularies/```
 - [x] Association Certainty
 - [x] Attestation Confidence
 - [x] Language and Script
 - [x] Name Accuracy
 - [x] Name Completeness
 - [x] Name Types
 - [x] Time Periods
 - [x] Feature (or Place) Categories
 - [x] Location Categories
 - [x] Connection Types
 - [ ] __Archaeological Remains (NOT VISIBLE!)__

## Verify that "hide" checkbox has expected effect

The expected effect is:
 - Any objects that previously had the hidden value __still__ have that value
 - Selection widgetry in the "edit" view for objects __does not__ include that value unless it is the pre-existing selected value (i.e., users can't select it)
 - HTML serialization of the vocabulary __includes__ the hidden values

The following vocabularies include the ability to "hide" terms (based on the affordances in the settings control panel):

 - [x] Time periods
 - [x] Place types
 - [x] Relationship types
 - [ ] __Association certainty (non-blocking criterion; DOES NOT WORK -- SEE #395)__
 - [ ] Attestation confidence (non-blocking criterion)
 - [ ] Language and script (non-blocking criterion)
 - [ ] Name types (non-blocking criterion)
 - [ ] Name accuracy (non-blocking criterion)
 - [ ] Name completeness (non-blocking criterion)

## Verify I can use advanced search or a collection criterion to find content items that have had a newly-added term selected
 - [x] Time Periods
 - [x] Feature (or Place) Categories
 - [ ] Archaeological Remains
 - [ ] Location Categories
 - [ ] Connection Types (aka Relationship Types)
 - [ ] Association Certainty
 - [ ] Attestation Confidence
 - [ ] Language and Script
 - [ ] Name Types
 - [ ] Name Accuracy
 - [ ] Name Completeness

## Make a list of content types that depend on each vocabulary and verify I can view, edit, and transition workflows successfully on each

@alecpm can you think of any other steps I should be taking to test this rigorously?

**Comment by alecpm** [comment]
I believe that's all that should be involved, but you would definitely know better than I do. There are some scripts like `Products/PleiadesEntity/skins/PleiadesEntity/getNameFacets.py` used for `Products/PleiadesEntity/skins/PleiadesEntity/browse_names.pt` that make use of the vocabs as well, I'm not sure where those appear on the site though.

**Comment by paregorios** [comment]
It looks like #394 may be related and so I've marked it as blocking progress on this ticket.

**Comment by alecpm** [comment]
Definitely related. I believe a fix is on its way to the staging server right now. I also opened up a [new PR on `pleiades.policy`](https://github.com/isawnyu/pleiades-policy/pull/2).

**Comment by alecpm** [comment]
This and #394 were due to some dumb copy/paste errors I made in my initial commits. These are now resolved and this is ready for testing on staging.

**Comment by alecpm** [comment]
This should no longer be blocked and is ready for review.

**Comment by alecpm** [comment]
All branches have been merged and deployed to production, and the migration was run.

**Comment by paregorios** [comment]
Evaluation on production: I cannot see an entry for "archaeological remains" on ```/vocabularies```. Based on the checklist above, I previously saw it on staging after the upgrade, so this looks perhaps like a missed step. I wanted to double-check staging, but it is timing out this morning.

**Comment by alecpm** [comment]
What's odd is the "archaeological remains" was added as a registry based vocabulary 3 years ago. I'm going to try running the old upgrade steps for that vocab to see if they put it in place, but I don't understand why it would have gone missing. I'll also take a look at staging.

**Comment by alecpm** [comment]
Looks like the vocabulary is there, but it's not included in the vocabulary view. You can see it by  searching the registry (https://pleiades.stoa.org/portal_registry) for "arch_remains". I'll add the vocab id to the listing view.

**Comment by alecpm** [comment]
It wasn't listed on staging either, but now it's in both. Turns out the vocabs that weren't originally ATVM vocabs need to be hardcoded into the custom listing view to appear there.

**Comment by paregorios** [comment]
Looks good. Closing as resolved.

**Comment by paregorios** [comment]
We never did the last bit, but we should: "Remove ATVocabularyManager from the Pleiades buildout. Hunt down and exterminate related cruft."
